File: lib/features/sample_module/application/actions/create_sample_action.dart
import 'package:codequest/features/sample_module/domain/entities/sample_item.dart';
import 'package:codequest/features/sample_module/domain/repositories/sample_repository_contract.dart';

class CreateSampleAction {
  CreateSampleAction(this._repository);

  final SampleRepositoryContract _repository;

  Future<SampleItem> call(String title) {
    if (title.trim().isEmpty) {
      throw ArgumentError('Título não pode ser vazio.');
    }
    return _repository.create(title: title.trim());
  }
}

File: firebase/functions/modules/sample/actions/create_sample_action.js
class CreateSampleAction {
  constructor(sampleRepository) {
    this.sampleRepository = sampleRepository;
  }

  async execute({ title }) {
    if (!title || !title.trim()) {
      throw new Error('Título é obrigatório.');
    }

    return this.sampleRepository.create({ title: title.trim() });
  }
}

module.exports = {
  CreateSampleAction,
};
